src/Spice86.Core/Emulator/Function/Dump/EmulatorStateSerializer.cs

```diff
@@ -13,33 +12,41 @@ public class EmulatorStateSerializer {
     private readonly IExecutionDumpFactory _executionDumpFactory;
     private readonly FunctionCatalogue _functionCatalogue;
     private readonly ILoggerService _loggerService;
-
     private readonly MemoryDataExporter _memoryDataExporter;
+    private readonly EmulatorBreakpointsSerializer _emulatorBreakpointsSerializer;
     
     /// <summary>
     /// Initializes a new instance of <see cref="EmulatorStateSerializer"/>.
     /// </summary>
-    public EmulatorStateSerializer(MemoryDataExporter memoryDataExporter, State state,
-        IExecutionDumpFactory executionDumpFactory, FunctionCatalogue functionCatalogue, ILoggerService loggerService) {
+    public EmulatorStateSerializer(
+        MemoryDataExporter memoryDataExporter, 
+        State state,
+        IExecutionDumpFactory executionDumpFactory,
+        EmulatorBreakpointsSerializer emulatorBreakpointsSerializer,
+        FunctionCatalogue functionCatalogue, 
+        ILoggerService loggerService) {
+        
         _state = state;
         _memoryDataExporter = memoryDataExporter;
         _executionDumpFactory = executionDumpFactory;
         _functionCatalogue = functionCatalogue;
         _loggerService = loggerService;
+        _emulatorBreakpointsSerializer = emulatorBreakpointsSerializer;
     }
     
-    
     /// <summary>
     /// Dumps the emulator state to the specified directory.
     /// </summary>
-    /// <param name="path">The directory used for dumping the emulator state.</param>
     public void SerializeEmulatorStateToDirectory(string path) {
         new RecordedDataWriter(
                 _state,
                 _executionDumpFactory,
                 _memoryDataExporter,
                 _functionCatalogue,
-                path, _loggerService)
+                path, 
+                _loggerService)
             .DumpAll();
+            
+        _emulatorBreakpointsSerializer.SaveBreakpoints();
```

Comment:
It's unexpected that this method also serializes the breakpoints, especially because the provided path is not considered during serialization.